src/global.F90

```diff
@@ -141,6 +141,7 @@ module global
   type(RegularMesh), allocatable, target :: meshes(:)
   type(TallyObject),    allocatable, target :: tallies(:)
   integer, allocatable :: matching_bins(:)
+  real(8), allocatable :: filter_weights(:)
```

Comment:
Can you brainstorm on how we can make these local variables rather than global threadprivate?